[#6] Implemented client API model with methods replacing logic in the…

robbieaverill opened the pull request:
Issue: #6 and #7 
### Resolution
1. Added Client_Api model
2. Added public methods `invalidate`, `validate`, `cleanup`, `reset` and `resetAndInvalidate`
3. Centralized logic using callbacks
4. Replaced logic from controller with API call
5. Handle array of table names or single table name
6. Changed "Invalidated" to "Validated" in the controller flash message (#7)

LeeSaferite commented on app/code/community/Aoe/EeIndexerStats/Model/Client/Api.php:
```diff
@@ -0,0 +1,123 @@
+            foreach ($tablenames as $tablename) {
+                /** @var Enterprise_Mview_Model_Client $client */
+                $client = Mage::getModel('enterprise_mview/client');
+                $client->initByTableName($tablename);
```
Isn't this subject to the same EE < 1.14.2 incompatibility?

LeeSaferite commented:
I know this is a few months old, but if you want to resolve the merge conflict I'll try to get it merged.

robbieaverill commented:
Sure, will do

robbieaverill commented:
Hi @LeeSaferite 

I've updated with the changes in master here, here are the changes from the original PR:
- Implement backward compatibility for EE into the API client
- Use the API in the observer to avoid duplication there too (and to provide EE compatibility as above)
- Fixed some formatting, doc blocks etc

Cheers,
Robbie

robbieaverill commented:
@LeeSaferite bump :)

robbieaverill commented:
@LeeSaferite bump :)

robbieaverill commented:
I'm closing this pull request to get it off my dashboard :) if you want the change you're welcome to cherry pick it from my fork.
